### PR TITLE
hostmetricsreceiver(filesystems): Refactor to use metrics metadata

### PR DESCRIPTION
**Pull request by asuresh4**
Refactor to use metrics metadata utilities.

### CODE REVIEW

File: receiver/hostmetricsreceiver/metadata.yaml [hunk 1]
```diff
@@ -14,6 +14,27 @@ labels:
...
 
+  filesystem.device:
+    value: device
+    description: Identifier of the filesystem.
+
+  filesystem.mode:
+    value: mode
+    description: Mountpoint mode such "ro", "rw", etc.
+
+  filesystem.mountpount:
```
Comment (jrcamp) [inline receiver/hostmetricsreceiver/metadata.yaml:25]: s/pount/point

File: receiver/hostmetricsreceiver/metadata.yaml [hunk 2]
```diff
@@ -50,3 +71,19 @@ metrics:
...
+  system.filesystem.usage:
+    description: Filesystem bytes used.
+    unit: By
+    data:
+      type: int sum
+      aggregation: cumulative
+      monotonic: false
+
+  system.filesystem.inodes.usage:
+    description: FileSystem iNodes used.
```
Comment (jrcamp) [inline receiver/hostmetricsreceiver/metadata.yaml:84]:
```suggestion
    description: FileSystem inodes used.
```

File: receiver/hostmetricsreceiver/metadata.yaml [hunk 3]
```diff
@@ -50,3 +71,19 @@ metrics:
...
+    unit: "{inodes}"
```
Comment (jrcamp) [inline receiver/hostmetricsreceiver/metadata.yaml:85]: I don't really know what the right unit is here but I've never seen one with braces.

```suggestion
    unit: "1"
```

Comment (asuresh4) [reply receiver/hostmetricsreceiver/metadata.yaml:85]: Actually, this is what we currently have. We have the same pattern in other places as well.

- [disk metrics](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.18.0/receiver/hostmetricsreceiver/internal/scraper/diskscraper/disk_metadata.go)
- [network metrics](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.18.0/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_metadata.go)
- [paging metrics](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.18.0/receiver/hostmetricsreceiver/internal/scraper/pagingscraper/paging_metadata.go)

@jrcamp Would you be ok if we address it separately and stick to what we currently have for now?

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@jrcamp please let me know if I can merge this

**Review by jrcamp — APPROVED** [review]

**Comment by jrcamp** [comment]
> @jrcamp please let me know if I can merge this

yes

**Pull request merged**